eeps/eep-0079.md

```diff
@@ -0,0 +1,784 @@
+
+```erlang
+Expr#_.Field
+```
+
+This access operation fails with a `{badfield,Field}` error if:
+
+* The field `Field` is not defined in the native-record value.
+
+### Updating native records
+
+The syntax for updating native-record values:
+
+```erlang
+Expr#Name{Field1=Expr1, ..., FieldN=ExprN}
+Expr#Module:Name{Field1=Expr1, ..., FieldN=ExprN}
+```
+
+Field names must be atoms.
+
+A native-record value is updated according to its native-record definition.
+
+An update operation fails with a `{badrecord,Expr}` error if:
+
+* The native-record definition is not visible at the call site (the
+  native record is not exported from `Module`).
+
+An update operation fails with a `{badfield,FN}` error if:
+
+* The native-record update expression references the field FN which is
+  not defined (in the structure definition).
+
+Native-record update expressions are not allowed in guards.
+
+### Anonymous update of native records
+
+The following syntax allows update any record having the given fields:
+
+```erlang
+Expr#_{Field1=Expr1, ..., FieldN=ExprN}
+```
+
+An anonymous update operation fails with a `{badfield,FN}` error if:
+
+* The native-record update expression references the field FN which is
+  not defined (in the structure definition).
+
+### Pattern matching over native records
+
+A pattern that matches a certain native-record value is created in the
+same way as a native-record is created.
+
+The syntax:
+
+```erlang
+#Name{Field1 = Expr1, ..., FieldN = ExprN}
+#Module:Name{Field1 = Expr1, ..., FieldN = ExprN}
+```
+
+Here, `Expr1`, .. `ExprN` are patterns, and Field names must be atoms.
+
+Pattern matching fails if:
+
+* The native-record value is not visible at the call site, that is,
+  an attempt is made to match out a a value from a non-exported record
+  from a module other than its defining module.
+
+* The pattern references a FieldK and the native-record value does not
+  contain this field.
+
+Note, however, that it is possible to match on the name of a
+non-exported record. Thus, if the `match_name/1` function in the
+following example is called with an instance of record `r` defined in
+`some_module`, it will succeed even if the record is not exported:
+
+```erlang
+-module(example).
+-export([match_name/1]).
+
+match_name(#some_module:r{}) ->
+  ok.
+```
+
+<!-- -->
+
+```erlang
+-module(some_module).
+-export([get_r/1]).
+
+-record #r{a}.
+
+get_r(N) -> #r{a=N}.
+```
+
+The following syntax allows matching any record having the named fields:
+
+```erlang
+#_{Field1 = Expr1, ..., FieldN = ExprN}
+```
+
+### Checking whether a record is current
+
+Do we need a way to check that the native-record is the current version?
+
+Yes, we should add a BIF that essentially does the following:
+
+```erlang
+record:get_fields(Record) =:= record:get_fields(#Module:Record{})
+```
+
+but guaranteed to always work and be more efficient.
+
+TODO: What should the name of the BIF be?
```

Comment:
Something like is_subtype or is_compatible, maybe.

Reply:
I think subtype and compatible would make me think the type of the fields are the same and are also checked. Perhaps `has_record_fields` but there are no guards starting with `has_`. I am not sure how to phrase it using `is_`.